packages/data-stores/src/domains/utils/assembler.ts

```diff
@@ -0,0 +1,163 @@
+import { camelCase, mapKeys } from 'lodash';
+import { DomainData } from '../../queries/use-site-domains-query';
+import {
+	DomainType,
+	GDPRConsentStatus,
+	GoogleEmailSubscription,
+	TitanEmailSubscription,
+	TransferStatus,
+} from '../types';
+import { getDomainType } from './get-domain-type';
+import { getGdprConsentStatus } from './get-gdpr-consent-status';
+import { getTransferStatus } from './get-transfer-status';
+
+function assembleGoogleAppsSubscription( googleAppsSubscription: {
+	status: string;
+	is_eligible_for_introductory_offer: boolean;
+} ) {
+	if ( ! googleAppsSubscription ) {
+		return;
+	}
+
+	return mapKeys( googleAppsSubscription, ( _, key ) => camelCase( key ) );
+}
+
+function assembleCurrentUserCannotAddEmailReason( reason: {
+	errors: { [ key: string ]: string[] };
+} ) {
+	if ( ! reason || ! reason.errors ) {
+		return null;
+	}
+
+	const errorDetails = Object.entries( reason.errors ).map( ( entry ) => {
+		const [ errorCode, errorMessages ] = entry;
+		return {
+			code: errorCode,
+			message: errorMessages[ 0 ],
+		};
+	} );
+	if ( ! errorDetails.length ) {
+		return null;
+	}
+	return errorDetails[ 0 ];
+}
+
+export const createSiteDomainObject = ( domain: DomainData ) => {
```

Comment:
It also does some transformation eg. there's no `transfer_end_date` on the domain api response, it's created in`createSiteDomainObject` by adding 7 days to the registration date. I think it make sense to have these transformations centralize instead of trying to applying them at various places that they might be needed.